### PR TITLE
Moved "ESH-INF/" folders to "OH-INF/" folders

### PR DESCRIPTION
**Pull request by cweitkamp**
- Moved `ESH-INF/` folders to `OH-INF/` folders

Related to https://github.com/openhab/openhab-core/pull/1328

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by wborn** [comment]
Since these changes won't be compatible with OH 2.5.x, I'd prefer to do one final SAT release that is compatible with OH 2.5.x. Preferably that release also has https://github.com/openhab/static-code-analysis/pull/365 so we can also significantly reduce the 2.5.x PR build times. WDYT @openhab/sat-maintainers?

**Comment by kaikreuzer** [comment]
Yes, yes, I wanted to look at #365 since a long time... Will try to review/merge it this weekend and do a new SAT release with it, ok?

**Comment by wborn** [comment]
Sounds good to me! :-)

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/EshInfXmlValidationCheck.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
  *
  */
 public class EshInfXmlValidationCheck extends AbstractEshInfXmlCheck {
```
Comment (kaikreuzer) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/EshInfXmlValidationCheck.java]: Shouldn't this class be renamed to `OhInfXmlValidationCheck`?

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractEshInfXmlCheck.java [hunk 2]
```diff
@@ -94,23 +94,23 @@ private void processXmlFile(FileText xmlFileText) throws CheckstyleException {
     }
 
     /**
-     * Validate a .xml file located in the ESH-INF/config directory
+     * Validate a .xml file located in the OH-INF/config directory
      *
      * @param xmlFileText - Represents the text contents of the xml file
      * @throws CheckstyleException when exception occurred during XML processing
      */
     protected abstract void checkConfigFile(FileText xmlFileText) throws CheckstyleException;
```
Comment (kaikreuzer) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractEshInfXmlCheck.java]: dito

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/CheckConstants.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
     // Directory names
     public static final String OSGI_INF_DIRECTORY_NAME = "OSGI-INF";
     public static final String META_INF_DIRECTORY_NAME = "META-INF";
-    public static final String ESH_INF_DIRECTORY = "ESH-INF";
+    public static final String ESH_INF_DIRECTORY = "OH-INF";
```
Comment (kaikreuzer) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/CheckConstants.java]: rename constant

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/CheckConstants.java [hunk 4]
```diff
@@ -38,7 +38,7 @@
...
     public static final String ESH_INF_PATH = "src/main/resources/" + ESH_INF_DIRECTORY;
```
Comment (kaikreuzer) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/api/CheckConstants.java]: rename constant

File: sat-plugin/src/test/java/org/openhab/tools/analysis/report/test/ReportUtilityTest.java [hunk 5]
```diff
@@ -107,7 +107,7 @@ public void assertWarningAreLoggedWhileExecuting() throws MojoFailureException {
         verify(logger).error("org.eclipse.smarthome.auth.jaas.internal.JaasAuthenticationProvider.java:[69]\nComment matches to-do format '(TODO)|(FIXME)'.");
```
Comment (kaikreuzer) [inline sat-plugin/src/test/java/org/openhab/tools/analysis/report/test/ReportUtilityTest.java]: Could we remove all `org.eclipse.smarthome` references as part of this PR as well?

**Comment by kaikreuzer** [comment]
@cweitkamp If you'd address the review comments, we should be good to merge.

**Comment by cweitkamp** [comment]
I must have missed those comments. Will take care of it soon.

**Comment by cweitkamp** [comment]
Done. I hope I found all occurrences.

**Pull request merged**